enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,467 @@
+
+As an alternative to using CNAME records, the ExternalDNS Route source implementation could point Route hostnames directly to
+LoadBalancer IP addresses or hostnames. This is not ideal since coordinating with administrator-managed load-balancers in this situation would be
+non-trivial. Using the `routerCanonicalHostname` field as a DNS middle-man (so to speak) simplifies ExternalDNS management.
+
+Note that the OpenShift [Route source implementation](https://github.com/kubernetes-sigs/external-dns/blob/master/source/ocproute.go)
+in ExternalDNS supports some of the common [ExternalDNS annotations](https://github.com/kubernetes-sigs/external-dns/blob/master/source/source.go),
+such as the ExternalDNS TTL annotation.
+
+Currently, the Ingress Operator publishes wildcard DNS Records to point an Ingress Controller's domain to the Ingress Controller's
+LoadBalancer type Service (or alternatively, a cluster administrator configures this for their administrator-managed Load Balancer, etc).
+In the future, the Ingress Operator could create DNSRecord Custom Resources to be managed by ExternalDNS (instead of the Ingress Operator's bespoke
+DNS provider). The Ingress Controller API could be expanded to include the ability for a cluster administrator to specify whether or not a Wildcard DNSRecord
+is desired.
+
+
+Here is an example of what an ExternalDNS deployment (created by the operator) would look like:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: external-dns
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      serviceAccountName: external-dns
+      containers:
+      - name: external-dns-private
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        - --policy=upsert-only # would prevent ExternalDNS from deleting any records, omit to enable full synchronization
+        # can only have one zone at a time if domains are overlapping over zones
+        - --aws-zone-type=private
+        - --registry=txt
+        - --txt-owner-id=externalDNS-operator
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7979
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          # Base64 decode from cloud-credentials secret generated by cloud credentials request
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+      - name: external-dns-public
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        - --policy=upsert-only # would prevent ExternalDNS from deleting any records, omit to enable full synchronization
```

Comment:
I would expect `--policy=upsert-only` would be removed to provide full management of the DNS record(s).

Reply:
Right. Fixed.